implementation/src/main/java/io/smallrye/mutiny/groups/MultiOnFailure.java

```diff
@@ -48,6 +52,43 @@ public MultiOnFailure(Multi<T> upstream, Predicate<? super Throwable> predicate)
                 null));
     }
 
+    /**
+     * Produces a new {@link Multi} invoking the given @{code action} when a {@code failure} event is received.
+     * <p>
+     * Unlike {@link #invoke(Consumer)}, the passed function returns a {@link Uni}. When the produced {@code Uni} sends
+     * its result, the result is discarded, and the original {@code failure} is forwarded downstream. If the produced
+     * {@code Uni} fails, a {@link io.smallrye.mutiny.CompositeException} composed by the original failure and the
+     * caught failure is propagated downstream.
+     * <p>
+     * If the asynchronous action throws an exception, this exception is propagated downstream as a
+     * {@link io.smallrye.mutiny.CompositeException} composed with the original failure and the caught exception.
+     * <p>
+     * This method preserves the order of the items, meaning that the downstream received the items in the same order
+     * as the upstream has emitted them.
+     *
+     * @param action the function taking the failure and returning a {@link Uni}, must not be {@code null}
+     * @return the new {@link Multi}
+     */
+    public Multi<T> invokeUni(Function<Throwable, ? extends Uni<?>> action) {
+        ParameterValidation.nonNull(action, "action");
+        return recoverWithMulti(failure -> {
+            Uni<?> uni = action.apply(failure);
+            if (uni == null) {
```

Comment:
Cosmetics

There and other groups: you could use `Objects. requireNonNull` instead.

Reply:
Yes, that's a good point. Let me at least update this one. We can replace the others in a "chore" PR.

Reply:
Done.